pylearn2/models/mlp.py

```diff
@@ -3272,16 +3342,14 @@ def fprop(self, state_below):
             p = self.output_normalization(p)
 
         return p
-
+       
+    @wraps(Layer.cost)
```

Comment:
Wrapping `Layer.cost` here is a good idea. But then, you should remove the part of the docstring containing the parameters and return type, since they are already in the docstring of `Layer.cost`. You could simply add a `Notes` section saying that the method simply calls `self.nonlin.cost`.